vllm/model_executor/layers/quantization/fp_eXmY.py

```diff
@@ -0,0 +1,203 @@
+from typing import Any, Dict, List, Optional
+
+import torch
+import torch.nn as nn
+
+from vllm import _custom_ops as ops
+from vllm.distributed import get_tensor_model_parallel_rank
+from vllm.logger import init_logger
+from vllm.model_executor.layers.linear import LinearBase, LinearMethodBase
+from vllm.model_executor.layers.quantization.base_config import (
+    QuantizationConfig)
+from vllm.model_executor.layers.quantization.utils.quant_llm_utils import (
+    _SPLIT_K_MAP, from_scaled_tc_fpx, to_scaled_tc_fpx)
+from vllm.model_executor.utils import set_weight_attrs
+
+logger = init_logger(__name__)
+
+# Used in vllm/config.py::ModelConfig::_verify_quantization
+VALID_FP_EXMY_METHODS = [
+    "fp4_weights", "fp5_weights", "fp6_weights", "fp7_weights"
+]
+DEFAULT_FP_EXMY_EXP_BITS = {
+    4: 2,
+    5: 2,
+    6: 2,
+    7: 3,
+}
+
+
+class QuantLLMFPConfig(QuantizationConfig):
+    """Config for QuantLLM FP quantizer. It supports fp4,
+    fp5, fp6, fp7.
+    
+    Reference: https://arxiv.org/abs/2401.14112
+    
+    Args: 
+        weight_bits: the target quantization bits, should be one of
+            4, 5, 6, 7.
+    """
+
+    def __init__(
+        self,
+        weight_bits: int = 6,
+        exp_bits: int = 2,
+    ) -> None:
+        self.weight_bits = weight_bits
+        self.exponent_bits = exp_bits
+
+        self.mantissa_bits = weight_bits - self.exponent_bits - 1
+
+        self.valid_types = [torch.float16]
+
+        if self.weight_bits not in DEFAULT_FP_EXMY_EXP_BITS:
```

Comment:
Could you also add a check that exp_bits is valid? It should be greater than 0 and less than 6 - right?